coderd/database/queries/workspaces.sql

```diff
@@ -687,3 +687,38 @@ UPDATE workspaces SET favorite = true WHERE id = @id;
 
 -- name: UnfavoriteWorkspace :exec
 UPDATE workspaces SET favorite = false WHERE id = @id;
+
+-- name: GetWorkspacesAndAgents :many
+SELECT
+	workspaces.id as id,
+	workspaces.name as name,
+	workspaces.owner_id as owner_id,
+	job_status,
+	transition,
+	(array_agg(ROW(agent_id, agent_name)::agent_id_name_pair) FILTER (WHERE agent_id IS NOT NULL))::agent_id_name_pair[] as agents
```

Comment:
Is there any particular reason you want to return these tuples instead of the full agent rows? I think the query should be renamed to `GetWorkspacesAndAgentNames` or something. Also on a big org this is going to return thousands of workspaces in a single query for owners...

Reply:
Yeah, I need to rethink how we can better use pubsub for this